logstash-core/lib/logstash/instrument/metric_type.rb

```diff
@@ -17,24 +17,23 @@
 
 require "logstash/instrument/metric_type/counter"
 require "logstash/instrument/metric_type/gauge"
-require "logstash/instrument/metric_type/uptime"
 
 module LogStash module Instrument
   module MetricType
-    METRIC_TYPE_LIST = {
-      :counter => LogStash::Instrument::MetricType::Counter,
-      :gauge   => LogStash::Instrument::MetricType::Gauge,
-      :uptime  => LogStash::Instrument::MetricType::Uptime,
-    }.freeze
-
     # Use the string to generate a concrete class for this metrics
     #
     # @param [String] The name of the class
     # @param [Array] Namespaces list
     # @param [String] The metric key
     # @raise [NameError] If the class is not found
     def self.create(type, namespaces, key)
-      METRIC_TYPE_LIST[type].new(namespaces, key)
+      case type
+      when :counter then return LogStash::Instrument::MetricType::Counter.new(namespaces, key)
+      when :gauge   then return LogStash::Instrument::MetricType::Gauge.new(namespaces, key)
+      when :uptime  then return org.logstash.instrument.metrics.UptimeMetric.new(key.to_s)
```

Comment:
review note: `UptimeMetric` doesn't need a ruby-based delegator, so it was removed.